ros2param/ros2param/api/__init__.py

```diff
@@ -13,113 +13,41 @@
-        value = parameter_value.bool_value
-    elif parameter_value.type == ParameterType.PARAMETER_INTEGER:
-        value = parameter_value.integer_value
-    elif parameter_value.type == ParameterType.PARAMETER_DOUBLE:
-        value = parameter_value.double_value
-    elif parameter_value.type == ParameterType.PARAMETER_STRING:
-        value = parameter_value.string_value
-    elif parameter_value.type == ParameterType.PARAMETER_BYTE_ARRAY:
-        value = list(parameter_value.byte_array_value)
-    elif parameter_value.type == ParameterType.PARAMETER_BOOL_ARRAY:
-        value = list(parameter_value.bool_array_value)
-    elif parameter_value.type == ParameterType.PARAMETER_INTEGER_ARRAY:
-        value = list(parameter_value.integer_array_value)
-    elif parameter_value.type == ParameterType.PARAMETER_DOUBLE_ARRAY:
-        value = list(parameter_value.double_array_value)
-    elif parameter_value.type == ParameterType.PARAMETER_STRING_ARRAY:
-        value = list(parameter_value.string_array_value)
-    elif parameter_value.type == ParameterType.PARAMETER_NOT_SET:
-        value = None
-    else:
-        value = None
-
-    return value
-
-
-def get_parameter_value(*, string_value):
-    """Guess the desired type of the parameter based on the string value."""
-    value = ParameterValue()
+    value = None
     try:
-        yaml_value = yaml.safe_load(string_value)
-    except yaml.parser.ParserError:
-        yaml_value = string_value
-
-    if isinstance(yaml_value, bool):
-        value.type = ParameterType.PARAMETER_BOOL
-        value.bool_value = yaml_value
-    elif isinstance(yaml_value, int):
-        value.type = ParameterType.PARAMETER_INTEGER
-        value.integer_value = yaml_value
-    elif isinstance(yaml_value, float):
-        value.type = ParameterType.PARAMETER_DOUBLE
-        value.double_value = yaml_value
-    elif isinstance(yaml_value, list):
-        if all((isinstance(v, bool) for v in yaml_value)):
-            value.type = ParameterType.PARAMETER_BOOL_ARRAY
-            value.bool_array_value = yaml_value
-        elif all((isinstance(v, int) for v in yaml_value)):
-            value.type = ParameterType.PARAMETER_INTEGER_ARRAY
-            value.integer_array_value = yaml_value
-        elif all((isinstance(v, float) for v in yaml_value)):
-            value.type = ParameterType.PARAMETER_DOUBLE_ARRAY
-            value.double_array_value = yaml_value
-        elif all((isinstance(v, str) for v in yaml_value)):
-            value.type = ParameterType.PARAMETER_STRING_ARRAY
-            value.string_array_value = yaml_value
-        else:
-            value.type = ParameterType.PARAMETER_STRING
-            value.string_value = string_value
-    else:
-        value.type = ParameterType.PARAMETER_STRING
-        value.string_value = yaml_value
+        value = parameter_value_to_python(parameter_value)
+    except RuntimeError as e:
+        print(f'Runtime error {e} raised')
```

Comment:
In the original implementation they ignored the error whereas in the `rclpy` implementation an error is raised. I think we should allow for the error to be raised but this is here to preserve the old behaviour

Reply:
Under what conditions do we expect to see this runtime error? I'm just wondering if we should quietly print nothing instead.

Reply:
Runtime error would occur if we hit an unexpected type. The only place where `get_value` is called in [`dump`](https://github.com/ros2/ros2cli/blob/faf3d052fc5bdc1cf981b5b1f49e5b9dc7b10148/ros2param/ros2param/verb/dump.py#L67) where I don't expect an unexpected type to ever appear.


I did it this way so that it matches the behaviour of the function before my refactor

Reply:
Makes sense to me!